src/daemon/bundle.rs

```diff
@@ -2,27 +2,35 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use crate::{
-    networks::{ActorBundleInfo, ACTOR_BUNDLES},
+    networks::{ActorBundleInfo, NetworkChain, ACTOR_BUNDLES},
     utils::{db::car_util::load_car, net::http_get},
 };
 use anyhow::ensure;
 use futures::{stream::FuturesUnordered, TryStreamExt};
 use fvm_ipld_blockstore::Blockstore;
 use std::io::Cursor;
+use std::mem::discriminant;
 use tracing::warn;
 
 /// Tries to load the missing actor bundles to the blockstore. If the bundle is
 /// not present, it will be downloaded.
-pub async fn load_actor_bundles(db: &impl Blockstore) -> anyhow::Result<()> {
+pub async fn load_actor_bundles(
+    db: &impl Blockstore,
+    network: &NetworkChain,
+) -> anyhow::Result<()> {
     FuturesUnordered::from_iter(
         ACTOR_BUNDLES
             .iter()
-            .filter(|bundle| !db.has(&bundle.manifest).unwrap_or(false))
+            .filter(|bundle| {
+                !db.has(&bundle.manifest).unwrap_or(false)
+                    && discriminant(network) == discriminant(&bundle.network)
```

Comment:
could you comment why you're using this instead of simply `==`?

Reply:
i.e is there a devnet bundle that's shared amongst all of them?

Reply:
Yeah. All devnets share the same set of bundles. I added a comment for clarity.